### PR TITLE
Fix scaling functions

### PR DESCRIPTION
**Pull request by f-fl0**
#### Description
This PR corrects two small errors that causes the improper results reported in the referenced issue:
- A copy paste typo (most likely) which sets the `Dy` component of the image rectangle to `Dx`.
- The `tmp` array in `fastBoxSampling` was not properly reset because the last argument in `memset` should be the number of bytes to fill in and not the number of elements.

#### Reference issue
Fixes https://github.com/pion/mediadevices/issues/312

### CODE REVIEW

**Review by at-wat — COMMENTED** [review]
Could you make the existing test fails on master branch and success on this branch?

**Comment by f-fl0** [comment]
I found another problem when using  the nearest neighbor scaling function when the same size as the input image:
image
 It is demonstrated by the test added in d56d4fb42127acf4b34b821c81aca55dc453f99b.

**Comment by f-fl0** [comment]
The need for the fix introduced in ee67dee4b812bb68e026cfb0b57115d9fb8a1b45 is demonstrated by the test added in 5f2216ace1313f2e521e9fe4b5ee0b82bff0f590.

**Comment by f-fl0** [comment]
The need for the fix introduced in ad6b0ad5e1c7878befe0dca92d82e09b3dae345d is demonstrated by the test added in edef5f5dee11be34eb8c12b719064b2764b6fd4d

**Comment by f-fl0** [comment]
@at-wat I added some tests and tried to demonstrate they would fail on master by reverting each change and let CI fail.

File: pkg/io/video/scale_test.go [hunk 1]
```diff
@@ -247,6 +525,103 @@ func TestScale(t *testing.T) {
...
+					0xD0, 0x98, 0x80,
+				},
+				YStride: 6,
+				CStride: 3,
+				Rect:    image.Rect(0, 0, 6, 4),
+			},
+		},
+	}
+
+	t.Run("FastBoxSampling", func(t *testing.T) {
```
Comment (at-wat) [inline pkg/io/video/scale_test.go:589]: This Run seems redundant (same as `func TestScaleFastBoxSampling(t *testing.T)`)

File: pkg/io/video/scale_test.go [hunk 2]
```diff
@@ -247,6 +525,103 @@ func TestScale(t *testing.T) {
 	}
 }
 
+func TestScaleFastBoxSampling(t *testing.T) {
+
```
Comment (at-wat) [inline pkg/io/video/scale_test.go:529]:
```suggestion
```

**Review by at-wat — COMMENTED** [review]
Overall looks good to me and added a few minor comments

File: pkg/io/video/scale_test.go (hunk 1 above)
Comment (f-fl0) [reply pkg/io/video/scale_test.go]: Removed in 466d14b1737a701bc31500700f8ef6f51147d76b.

File: pkg/io/video/scale_test.go (hunk 2 above)
Comment (f-fl0) [reply pkg/io/video/scale_test.go]: Removed in 466d14b1737a701bc31500700f8ef6f51147d76b.

**Review by at-wat — APPROVED** [review]
LGTM

**Pull request merged**